src/python/grpcio/grpc/_cython/_cygrpc/aio/call.pyx.pxi

```diff
@@ -327,7 +402,31 @@ cdef class _AioCall(GrpcCallWrapper):
         metadata_sent_observer()
 
         # Receives initial metadata.
-        initial_metadata_observer(
-            await _receive_initial_metadata(self,
-                                            self._loop),
+        self._set_initial_metadata(
+            await _receive_initial_metadata(self, self._loop)
         )
+
+
+cdef _AioCall new_AioCall(AioChannel channel, object deadline,
```

Comment:
If instantiation of these `_AioCall` objects is one of the bottlenecks, perhaps we would benefit from a [freelist](https://cython.readthedocs.io/en/latest/src/userguide/extension_types.html#fast-instantiation)?

Reply:
I didn't know about the freelist, indeed I was thinking that how we could do preallocation. Thanks for sharing it.

But in any case, I ended up by making some tests and it turned out that with the constructor proposed in this PR, 90% of the time is spent in the `_create_grpc_call`, presumably in the `grpc_channel_create_call` which for me was a big surprise. I can share later more numbers about this.

Reply:
Core uses arena allocation behind the scenes. As you have more queries in flight, it will expand the size of the arena (I believe it uses doubling). So your first first queries will take the hit of that allocation. Perhaps we could do some tuning in core to start the arena out with a bigger initial size. That should be equivalent to adding a free list to the memory backing `grpc_channel_create_call`.